langchain4j/src/main/java/dev/langchain4j/service/output/JsonSchemas.java

```diff
@@ -32,20 +38,66 @@ public static Optional<JsonSchema> jsonSchemaFrom(Type returnType) {
             throw illegalConfiguration("Return type of method '%s' cannot be void");
         }
 
-        if (!isPojo(returnType)) {
+        if (!isPojo(returnType) && !isEnum(returnType)) {
             return Optional.empty();
         }
 
         Class<?> rawClass = getRawClass(returnType);
 
-        JsonSchema jsonSchema = JsonSchema.builder()
-                .name(rawClass.getSimpleName())
-                .rootElement(jsonObjectOrReferenceSchemaFrom(rawClass, null, new LinkedHashMap<>(), true))
-                .build();
+        JsonSchema jsonSchema;
+
+        if (typeHasRawClass(returnType, List.class) || typeHasRawClass(returnType, Set.class)) {
+            rawClass = resolveFirstGenericParameterClass(returnType);
+            Class<?> finalRawClass = rawClass;
+            if (finalRawClass != null && finalRawClass.isEnum()) {
+                jsonSchema = JsonSchema.builder()
+                        .name(getRawClass(returnType).getSimpleName() + "_of_" + rawClass.getSimpleName())
+                        .rootElement(JsonObjectSchema.builder()
+                                .addProperty("array", JsonArraySchema.builder()
```

Comment:
Should have a more descriptive name instead of `array`?

Reply:
changed to "items"